samcli/hook_packages/terraform/hooks/prepare/resources/internal.py

```diff
@@ -0,0 +1,15 @@
+"""Module containing logic specific to internal resources handling during the prepare hook execution"""
+from samcli.hook_packages.terraform.hooks.prepare.types import ResourceProperties
+
+INTERNAL_PREFIX = "Internal::"
```

Comment:
is there a way where we can prevent accidental injection of this temporary resource into metadata?

Reply:
Other than this condition https://github.com/aws/aws-sam-cli/blob/1d57965b88139076d686d8e804818876b4fe577c/samcli/hook_packages/terraform/hooks/prepare/translate.py#L225 I'm not sure what else we can do. Maybe a post-processing step to ensure we didn't write anything we weren't supposed to? Though even if we did write it to the file, I don't think it should break local behaviour.

Reply:
I am not sure if we do validate the Template somewhere in SAM CLI while processing it or not, if we do so adding this internal type by mistake will cause issues.

Anyway, we only have one place now where we add the resources to Metadata file, and we guard it with this condition, we may face this issue if we start updating the Metadata file from different places. I think for now we can keep it simple.

Reply:
Agree not blocking. 

My concern is that this resource should never accidentally make it to the template itself as it may fail certain aspects? Maybe add a TODO note.

Reply:
I added a TODO and also confirmed that if we accidentally do add an internal resources to the metadata file then the local test suite still works.